File: 0006-zigzag-conversion/0006-zigzag-conversion.java
class Solution {
    public String convert(String s, int numRows) {
        if(numRows==1 ||numRows>=s.length()){
            return s;
        }
        StringBuilder [] rows = new StringBuilder[numRows];
        for(int i=0;i<numRows;i++){
            rows[i]=new StringBuilder();
        }
        int index=0;
        int step=1;
        for(char c :s.toCharArray()){
            rows[index].append(c);
            if(index==0){
                step=1;
            }
            else if(index==numRows-1){
                step=-1;
            }
            index+=step;
        }
        StringBuilder result= new StringBuilder();
        for(StringBuilder row: rows){
            result.append(row);
        }
        return result.toString();
        
        
        
    }
}

File: 0006-zigzag-conversion/0006-zigzag-conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows==1 or numRows>=len(s):
            return s
        rows=['']*numRows
        index,step=0,1
        for char in s:
            rows[index]+=char
            if index==0:
                step=1
            elif index==numRows-1:
                step=-1
            index+=step
        return ''.join(rows)
        
        
        




#Better Approach
# def convert(s: str, numRows: int) -> str:
#     if numRows == 1 or numRows >= len(s):
#         return s

#     result = ''
#     index, step = 0, 1

#     for char in s:
#         result += char
#         if index == 0:
#             step = 1
#         elif index == numRows - 1:
#             step = -1
#         index += step

#     return result